### PR TITLE
feat: oui skeleton

### PR DESCRIPTION
**Pull request by varun257**
add oui skeleton component
add examples and test case

cui skeleton to oui

JIRA: UU-845

### CODE REVIEW

File: packages/oui-skeleton/src/skeleton.component.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import controller from "./skeleton.controller";
+import template from "./skeleton.html";
+
+export default {
+    controller,
+    controllerAs: "$ctrl",
```
Comment (radireddy) [inline packages/oui-skeleton/src/skeleton.component.js]: this is the default value, you can remove it

File: packages/oui-skeleton/src/skeleton.html [hunk 2]
```diff
@@ -0,0 +1,6 @@
+<div class="oui-skeleton">
+    <span class="oui-skeleton__loader"
+      ng-show="$ctrl.loading"
+      data-ng-style="{width: $ctrl.width}"></span>
+    <div ng-hide="$ctrl.loading" ng-transclude></div>
```
Comment (radireddy) [inline packages/oui-skeleton/src/skeleton.html]: use ng-if

File: packages/oui-skeleton/src/skeleton.controller.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import { addDefaultParameter } from "@oui-angular/common/component-utils";
+
+export default class {
+    constructor ($attrs) {
+        "ngInject";
+
+        this.$attrs = $attrs;
+        this.max = 100;
```
[the remark this answers is not in the document]
Comment (varun257) [reply packages/oui-skeleton/src/skeleton.controller.js]: Yup this should be good. Better than choosing random width. Will do this

File: packages/oui-skeleton/README.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+<component-status cx-design="complete" ux="rc"></component-status>
+
+## Usage
+
+### Default
+```html:preview
+<oui-skeleton></oui-skeleton>
+```
+
+### Size
```
Comment (neolitec) [inline packages/oui-skeleton/README.md]: Maybe size is not the thing we want.
Typically the skeleton graphically represents a text we don't know in advance. So the size should be... randomized :)
Maybe the right parameters here should be "min-width" and eventually "width" / "auto" if we want to force the width... ?
@AxelPeter what do you think about?

Comment (AxelPeter) [reply packages/oui-skeleton/README.md]: Nice idea. But I don't think it's the case for all uses.
We can even have both with an attribute "randomized" for example :)

In fact, I'm not a fan of replacing `style="width: 80%"` just by `width="80%"`. That's why by default, I let the skeleton with `auto` to let the user set its width.

I'll check to add your idea

**Pull request merged**